packages/analytics-js-plugins/src/errorReporting/utils.ts

```diff
@@ -0,0 +1,182 @@
+  SDK_GITHUB_URL,
+  NOTIFIER_NAME,
+  SOURCE_NAME,
+} from './constants';
+import { json } from '../shared-chunks/common';
+
+const getConfigForPayloadCreation = (err: SDKError, errorType: string) => {
+  switch (errorType) {
+    case 'unhandledException': {
+      const { error } = err as ErrorEvent;
+      return {
+        component: 'unhandledException handler',
+        tolerateNonErrors: true,
+        errorFramesToSkip: 1,
+        normalizedError: error,
+      };
+    }
+    case 'unhandledPromiseRejection': {
+      const error = err as PromiseRejectionEvent;
+      return {
+        component: 'unhandledrejection handler',
+        tolerateNonErrors: false,
+        errorFramesToSkip: 1,
+        normalizedError: error.reason,
+      };
+    }
+    case 'handledException':
+    default:
+      return {
+        component: 'notify()',
+        tolerateNonErrors: true,
+        errorFramesToSkip: 2,
+        normalizedError: err,
+      };
+  }
+};
+
+const createNewBreadCrumb = (message: string, metaData?: BreadCrumbMetaData): BreadCrumb => ({
+  type: 'manual',
+  message,
+  timestamp: new Date(),
+  metadata: metaData || {},
+});
+
+const getReleaseStage = () => {
+  const host = globalThis.location.hostname;
+  return host && DEV_HOSTS.includes(host) ? 'development' : '__RS_BUGSNAG_RELEASE_STAGE__';
+};
+
+const getAppStateForMetadata = (state: ApplicationState): Record<string, any> => {
+  const stateStr = json.stringifyWithoutCircular(state, false, APP_STATE_EXCLUDE_KEYS);
+  return stateStr !== null ? JSON.parse(stateStr) : {};
+};
+
+const getURLWithoutSearchParam = () => {
+  const url = globalThis.location.href.split('?');
+  return url[0];
+};
+
+const getErrorContext = (event: any) => {
+  const { errorMessage } = event;
+  let context = errorMessage;
+
+  // Hack for easily grouping the script load errors
+  // on the dashboard
+  if (errorMessage.includes('error in script loading')) {
+    context = 'Script load failures';
+  }
+  return context;
+};
+
+const enhanceErrorEvent = (
+  payload: Event,
+  errorState: ErrorState,
+  state: ApplicationState,
+): ErrorEventPayload => ({
+  notifier: {
+    name: NOTIFIER_NAME,
+    version: '__PACKAGE_VERSION__',
+    url: SDK_GITHUB_URL,
+  },
+  events: [
+    {
+      payloadVersion: '5',
+      exceptions: clone(payload.errors),
+      severity: errorState.severity,
+      unhandled: errorState.unhandled,
+      severityReason: errorState.severityReason,
+      app: {
+        version: '__PACKAGE_VERSION__',
+        releaseStage: getReleaseStage(),
+      },
+      device: {
+        locale: state.context.locale.value || undefined,
+        userAgent: state.context.userAgent.value || undefined,
+        time: new Date(),
+      },
+      request: {
+        url: getURLWithoutSearchParam() as string,
+        clientIp: '[NOT COLLECTED]',
+      },
+      breadcrumbs: payload.breadcrumbs,
+      context: getErrorContext(payload.errors[0]),
+      metaData: {
```

Comment:
Yes, I intentionally didn't add it. Previously we used to send only sourceId to find the source. Since we have the writekey, again having the sourceId doesn't make much sense. We can find it using writekey.
If you think otherwise, we can add it.

Reply:
Not just the write key, but this also needs the snippet version that was added recently.

The source write key is not meant for purposes like this. Hence, we also used source ID previously.

Reply:
Addressed